crates/bifrost/src/watchdog.rs

```diff
@@ -71,39 +82,96 @@ impl Watchdog {
         let bifrost = self.inner.clone();
-        let _ = TaskCenter::spawn_child(
+        TaskCenter::spawn_unmanaged(
             TaskKind::BifrostBackgroundLowPriority,
-            format!("trim-chain-{log_id}"),
+            "trim-chains",
             async move {
-                let trim_point = bifrost
-                    .get_trim_point(log_id)
-                    .await
-                    .context("cannot determine tail")?;
-                if trim_point == Lsn::INVALID {
-                    return Ok(());
+                if trim_requests.is_empty() {
+                    return trim_requests;
+                }
+
+                let mut trim_points = Vec::with_capacity(trim_requests.len());
+
+                // Concurrently look up the trim points of all the requests
+                let mut trim_point_futures: FuturesUnordered<_> = trim_requests
+                    .drain()
+                    .map(async |(log_id, requested_trim_point)| {
+                        match bifrost.get_trim_point(log_id).await {
+                            // an invalid lsn means that the log has never been trimmed
+                            Ok(actual_trim_point) if actual_trim_point == Lsn::INVALID => Ok(None),
+                            Ok(actual_trim_point) => Ok(Some(TrimPoint {
+                                log_id,
+                                requested_trim_point,
+                                actual_trim_point,
+                            })),
+                            Err(_) => Err((log_id, requested_trim_point)),
+                        }
+                    })
+                    .collect();
+
+                let mut failed_trim_requests = trim_requests;
+                debug_assert!(failed_trim_requests.is_empty());
+
+                while let Some(trim_point_result) = trim_point_futures.next().await {
+                    match trim_point_result {
+                        Ok(Some(trim_point)) => {
+                            trim_points.push(trim_point);
+                        }
+                        // skip invalid lsns
+                        Ok(None) => continue,
+                        Err((log_id, requested_trim_point)) => {
+                            // keep track of requests we failed to find trim points for
+                            failed_trim_requests.insert(log_id, requested_trim_point);
+                        }
+                    }
+                }
+
+                if trim_points.is_empty() {
+                    if !failed_trim_requests.is_empty() {
+                        warn!(
+                            "Bifrost watchdog trim chains failed; could not retrieve any trim points"
+                        );
+                    }
+                    return failed_trim_requests;
                 }
+
                 // todo(asoli): Notify providers about trimmed loglets for pruning.
                 let opts = Configuration::pinned();
```

Comment:
This might keep the pinned configuration across an await point. `let retry_policy = Configuration::pinned().common.network_error_retry_policy.clone();` should free the pinned configuration immediately.